socorro/external/postgresql/connection_context.py

```diff
@@ -49,6 +53,13 @@ class ConnectionContext(RequiredConfig):
         reference_value_from='secrets.postgresql',
         secret=True,
     )
+    required_config.add_option(
+        name='database_url',
+        default=None,
+        doc='URI for connecting to the database, '
+            'format: postgres://username:pass@hostname:port/database',
```

Comment:
What about other things like `?sslmode=verify` or something like that? Or `&client_encoding=utf8`. I'm sort of making these up but my point is that there might be additional query string parameters there that someone might want to send in to the `psycopg2.connect(**options)` thing. 
Perhaps this is not the time and place but I think if we refactor ConnectionContext further I'd like to make it stop using a DSN string and instead unpack everything in this URI into the keyword arguments form for `psycopg2.connect()`.